### PR TITLE
Give `t5` the `prune_heads`

### PR DESCRIPTION
**Pull request by CaffreyR**
# What does this PR do?
Give `t5` the `prune_heads`  #19975

Fixes # (issue #19625)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@ArthurZucker @patrickvonplaten @patil-suraj
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by CaffreyR** [comment]
Hi @ArthurZucker ,let's try this one!

**Comment by CaffreyR** [comment]
Hi @ArthurZucker , I have passes the test!

**Comment by CaffreyR** [comment]
Hi @ArthurZucker , this version is clean. Could you please give it a review?

**Comment by CaffreyR** [comment]
Gently Pin @patrickvonplaten 
Maybe you are also interested.

**Comment by CaffreyR** [comment]
Hi @ArthurZucker , could you please give me a review? Many thanks!

File: docs/source/en/contributing.md [hunk 1]
```diff
@@ -1 +0,0 @@
-../../../CONTRIBUTING.md
```
Comment (patrickvonplaten) [inline docs/source/en/contributing.md:1]: Let's please not delete this

**Review by patrickvonplaten — COMMENTED** [review]
Thanks for the PR! 
We'll need tests before we can merge this.

Could you set this to `True`: https://github.com/huggingface/transformers/blob/07b8f249cdb07a5e6697b379cc6db705a9eb15f1/tests/models/t5/test_modeling_t5.py#L521 

and then see if the tests all pass?

**Comment by CaffreyR** [comment]
> Thanks for the PR! We'll need tests before we can merge this.
> 
> Could you set this to `True`:
> 
> https://github.com/huggingface/transformers/blob/07b8f249cdb07a5e6697b379cc6db705a9eb15f1/tests/models/t5/test_modeling_t5.py#L521
> 
> and then see if the tests all pass?

Sure, I modify it manually in my repo

**Comment by CaffreyR** [comment]
Hi @patrickvonplaten , it seems some of the test failed, because I did not modify the `T5stack` but `T5forconditionalgeneration`

**Comment by patrickvonplaten** [comment]
> Hi @patrickvonplaten , it seems some of the test failed, because I did not modify the `T5stack` but `T5forconditionalgeneration`

Could you try to make the tests work - we need those to pass before we're able to merge this PR :-)

**Pull request closed without merging**